crates/iceberg/src/arrow/record_batch_transformer.rs

```diff
@@ -83,14 +83,10 @@ fn constants_map(
             // Handle both None (null) and Some(Literal::Primitive) cases
             match &partition_data[pos] {
                 None => {
-                    // TODO (https://github.com/apache/iceberg-rust/issues/1914): Add support for null datum values.
-                    return Err(Error::new(
-                        ErrorKind::Unexpected,
-                        format!(
-                            "Partition field {} has null value for identity transform",
-                            field.source_id
-                        ),
-                    ));
+                    // Skip null partition values - they will be resolved as null per Iceberg spec rule #4.
```

Comment:
I got it, it's a little tricky. When it's value is null, it will be resolved to null since it's optional.

Reply:
I think there is a corner case where this approach will fail, as spec says identity partition should override values from underlying files, with this fix we will still return values from underlying files rather than null identity partition value.